### ISSUE TITLE
switch to c++11

### ISSUE THREAD

**Issue by rwcarlsen** [cyclus/cyclus#1213]
newest versions of libxml++ require c++ 11 - i.e. our build will be broken without the ``-std=c++11`` flag.

**Comment by gidden** [comment]
There's a PR in the hopper for this:
https://github.com/cyclus/cyclus/pull/1118

**Comment by rwcarlsen** [comment]
Looks like pyne could use a few explicit casts - it won't compile under c++11:

```bash
/home/r/src/cyclus/cyclus/src/pyne.cc:13435:29: error: non-constant-expression cannot be narrowed from type 'int' to 'hsize_t'
      (aka 'unsigned long long') in initializer list [-Wc++11-narrowing]
  hsize_t data_offset[1] = {row};
                            ^~~
/home/r/src/cyclus/cyclus/src/pyne.cc:13435:29: note: insert an explicit cast to silence this issue
  hsize_t data_offset[1] = {row};
                            ^~~
                            static_cast<hsize_t>( )
/home/r/src/cyclus/cyclus/src/pyne.cc:13459:26: error: non-constant-expression cannot be narrowed from type 'int' to 'hsize_t'
      (aka 'unsigned long long') in initializer list [-Wc++11-narrowing]
  hsize_t nuc_dims[1] = {nuc_size};
                         ^~~~~~~~
/home/r/src/cyclus/cyclus/src/pyne.cc:13459:26: note: insert an explicit cast to silence this issue
  hsize_t nuc_dims[1] = {nuc_size};
                         ^~~~~~~~
                         static_cast<hsize_t>( )
/home/r/src/cyclus/cyclus/src/pyne.cc:13711:29: error: non-constant-expression cannot be narrowed from type 'int' to 'hsize_t'
      (aka 'unsigned long long') in initializer list [-Wc++11-narrowing]
    hsize_t chunk_dims[1] ={chunksize};
                            ^~~~~~~~~
/home/r/src/cyclus/cyclus/src/pyne.cc:13711:29: note: insert an explicit cast to silence this issue
    hsize_t chunk_dims[1] ={chunksize};
                            ^~~~~~~~~
                            static_cast<hsize_t>( )
```

**Comment by rwcarlsen** [comment]
fixed by #1216.
